base/linalg/symmetric.jl

```diff
@@ -266,9 +266,9 @@ for (funm, func) in ([:logm,:log], [:sqrtm,:sqrt])
         function ($funm)(A::Symmetric)
             F = eigfact(A)
             if isposdef(F)
-                retmat = (F.vectors * Diagonal(($func)(F.values))) * F.vectors'
+                retmat = (F.vectors * Diagonal(($func).(F.values))) * F.vectors'
             else
-                retmat = (F.vectors * Diagonal(($func)(complex(F.values)))) * F.vectors'
+                retmat = (F.vectors * Diagonal(($func).(complex(F.values)))) * F.vectors'
```

Comment:
> Shouldn't we deprecate `float` and `complex` and `big` for arrays in favor of dot calls?

Absolutely. This PR attacks only macro-vectorized methods, leaving manually vectorized methods for a followup PR. (My approach has been to select a macro-vectorized method, deprecate it, find its usages via depwarns, migrate those usages, rinse and repeat. As such I presently miss, e.g., `float` calls that dispatch to manually vectorized methods. But I've been accumulating a list of manually vectorized functions to attack in the followup PR and much look forward to diving in.)